src/server/game/Entities/Player/CollectionMgr.cpp

```diff
@@ -259,3 +260,137 @@ bool CollectionMgr::CanApplyHeirloomXpBonus(uint32 itemId, uint32 level)
 
     return level <= 60;
 }
+
+MountDefinitionMap CollectionMgr::_mountDefinitions;
+
+void CollectionMgr::LoadAccountMounts(PreparedQueryResult result)
+{
+    if (!result)
+        return;
+
+    do
+    {
+        Field* fields = result->Fetch();
+        uint32 spellId = fields[0].GetUInt32();
+        bool isFavourite = fields[1].GetBool();
+
+        if (!sDB2Manager.GetMount(spellId))
+            continue;
+
+        _mounts[spellId] = isFavourite;
+    } while (result->NextRow());
+}
+
+void CollectionMgr::SaveAccountMounts(SQLTransaction& trans)
+{
+    PreparedStatement* stmt = nullptr;
+    for (auto const& mount : _mounts)
+    {
+        stmt = LoginDatabase.GetPreparedStatement(LOGIN_REP_ACCOUNT_MOUNTS);
+        stmt->setUInt32(0, _owner->GetBattlenetAccountId());
+        stmt->setUInt32(1, mount.first);
+        stmt->setBool(2, mount.second);
+        trans->Append(stmt);
+    }
+}
+
+bool CollectionMgr::UpdateAccountMounts(uint32 spellId, bool isFavourite)
+{
+    return _mounts.insert(MountContainer::value_type(spellId, isFavourite)).second;
```

Comment:
_mounts.insert(MountContainer::value_type(spellId, isFavourite).second);

Maybe because of this, mounts are not being updated upon learning?
Not sure tho..